pkg/controller/namespace/namespace_controller.go

```diff
@@ -132,7 +132,9 @@ func (nm *NamespaceController) enqueueNamespace(obj interface{}) {
 		glog.Errorf("Couldn't get key for object %+v: %v", obj, err)
 		return
 	}
-	nm.queue.Add(key)
+	// delay processing namespace deletions long enough for the lifecycle admission
+	// plugins to also observe the deletion and stop new objects from being created
+	nm.queue.AddAfter(key, 5*time.Second)
```

Comment:
> Why admission doesn't reject new objects for namespaces that have the deletion timestamp set?

It does, but in an HA apiserver scenario, knowledge that a namespace has a deletionTimestamp set takes non-zero time to propagate, so apiserver B can allow a create after a namespace was deleted via apiserver A